cmd/goal/clerk.go

```diff
@@ -724,46 +724,77 @@ var signCmd = &cobra.Command{
 
 		var outData []byte
 		dec := protocol.NewDecoderBytes(data)
+		// read the entire file and prepare in-memory copy of each signed transaction, with grouping.
+		txnGroups := make(map[crypto.Digest][]*transactions.SignedTxn)
+		groupsOrder := make([]crypto.Digest, 0)
```

Comment:
don't need to `make()`, `append()` and `range` will do the right thing with a nil array; can just `var groupsOrder []crypto.Digest`; OR preallocate so the append()s go faster `make([]T, 0, 5)`

Reply:
good point; however, in this case where I can't tell ( easily ) ahead of time, I'll just use
```
var groupsOrder []crypto.Digest
```